### PR TITLE
Use document metadata during (re)indexing

### PR DESCRIPTION
**Pull request by cwang**
## Description

Adding document metadata during (re)indexing to support https://github.com/docqai/docq/pull/36

cc @osala-eng

Fixes/Implements #(issue/feature)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Refactor and code improvement (non-breaking change which improves code quality and/or performance)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation
- [ ] Tests
- [ ] Other chores such as maintenance

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration by modifying the list below.

- [ ] Test A
- [ ] Test B
- [ ] Test C

**Test Configuration**:

Please describe the test setup. List them below as bullet points.

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] I have checked my code and corrected any misspellings
- [ ] The commit message follows the convention of this project

### CODE REVIEW

File: source/docq/config.py [hunk 1]
```diff
@@ -12,6 +13,14 @@
 }
 
 
+class DocumentMetadata(Enum):
+    """Document metadata."""
+
+    FILE_PATH = "file_path"
```
Comment (janaka) [inline source/docq/config.py:19]: are you assuming this is a disk path or can it be a uri? 

other than manually uploaded docs right now I think all other source will be referencing a location off disk.

Comment (cwang) [reply source/docq/config.py:19]: Still some kind of *path* I guess? So `file_path` is the path to a file, not the path in a file system :)

Comment (janaka) [reply source/docq/config.py:19]: yep, I don't have an issue with the name itself. More syncing on your thinking 👍🏾

**Review by janaka — APPROVED** [review]

File: source/docq/config.py [hunk 2]
```diff
@@ -3,6 +3,7 @@
 from enum import Enum
 
 ENV_VAR_DOCQ_DATA = "DOCQ_DATA"
+ENV_VAR_DOCQ_DEMO = "DOCQ_DEMO"
```
Comment (cwang) [inline source/docq/config.py:6]: @osala-eng FYI use this as the demo flag

Comment (osala-eng) [reply source/docq/config.py:6]: Okay

**Pull request merged**